latest/index.bs

```diff
@@ -263,16 +261,17 @@ Each dictionary in "datasets" MUST contain the field "path", whose value contain
 to the current zarr group. The "path"s MUST be ordered from largest (i.e. highest resolution) to smallest.
 
 Each "datasets" dictionary MUST have the same number of dimensions and MUST NOT have more than 5 dimensions. The number of dimensions and order MUST correspond to number and order of "axes".
-Each dictionary MAY contain the field "transformations", which contains a list of transformations that map the data coordinates to the physical coordinates (as specified by "axes") for this resolution level.
-The transformations are defined according to [[#trafo-md]]. In addition, the transformation types MUST only be `identity`, `translation` or `scale`.
-They MUST contain at most one `scale` transformation per axis that specifies the pixel size in physical units.
-It also MUST contain at most one `translation` per axis that specifies the offset from the origin in physical units.
-If both `scale` and `translation` are given `translation` must be listed after `scale` to ensure that it is given in physical coordinates. If "transformations" is not given, the identity transformation is assumed.
+Each dictionary MAY contain the field "coordinateTransformations", which contains a list of transformations that map the data coordinates to the physical coordinates (as specified by "axes") for this resolution level.
+The transformations are defined according to [[#trafo-md]]. The transformation types MUST only be `identity`, `translation` or `scale`.
+They MUST contain at most one `scale` transformation that specifies the pixel size in physical units or time duration.
+It also MUST contain at most one `translation` that specifies the offset from the origin in physical units.
+The length of the `scale` and/or `translation` array MUST be the same as the length of "axes".
+If both `scale` and `translation` are given `translation` must be listed after `scale` to ensure that it is given in physical coordinates. If "coordinateTransformations" is not given, the identity transformation is assumed.
 The requirements (only `scale` and `translation`, restrictions on order) are in place to provide a simple mapping from data coordinates to physical coordinates while
 being compatible with the general transformation spec.
 
-Each "multiscales" dictionary MAY contain the field "transformations", describing transformations that are applied to each resolution level.
-The transformations MUST follow the same rules about allowed types, order, etc. as in "datasets:transformations".
+Each "multiscales" dictionary MAY contain the field "coordinateTransformations", describing transformations that are applied to each resolution level.
```

Comment:
I suggest we remove this 'global' transformation in order to keep the proposal simple. (and this additional transformation would not make the spec more expressive)

Reply:
What are the right assumptions to make if the "multiscales" does not contain a `coordinateTransformations`?
Assume downsampled by 2 in all dimensions always? infer from dataset sizes?

Reply:
I would really hope the answer is "fail".

Reply:
> What are the right assumptions to make if the "multiscales" does not contain a coordinateTransformations?
Assume downsampled by 2 in all dimensions always? infer from dataset sizes?

Good point. In the previous versions we have infered from the dataset size because this was the only viable option.
But that was a very hacky solution, so I agree with @jbms that we can also say "Each multiscales dictionary MUST contain ..." instead of "... MAY contain ...". Any thoughts @will-moore @joshmoore @sbesson?

Reply:
In `napari` and `vizarr` I don't know a way to use transformations on each datasets level - I just pass in the pyramid and they make the assumption that each level covers the same area. In fact, `vizarr` even assumes the scaling factor is 2 between each level. 
I can only use a single `scale` or `translate` transform - I don't know if it makes a difference whether that is on the first `datasets` dict or on the `multiscales` dict.
So I think that it is a valid default behaviour to infer from the size of each level, and if you want the default behaviour then I don't think we MUST need `coordinateTransformations` on every datasets level or even on the `multiscales` dict.

Reply:
It seems to me that whatever default is chosen, if it is allowed that the downsampling factors and offsets be left unspecified, it will be non-obvious what the default is, at least to a significant fraction of users, which will lead to users either guessing wrong, or having to look up the spec every time they encounter an unspecified trnasformation.  I would suppose that in the case of an xy dataset where the dimensions are powers of 2, you will probably guess correctly, but if the sizes are not exactly divisible by 2, or there are more than two dimensions, there is a lot of room for guessing wrong.  Even worse than users guessing wrong, there is also a risk of implementation divergence.  If the downsampling factors and offsets are specified explicitly, that problem is avoided, and it is not very difficult to specify the factors explicitly when writing.

Regarding the issue with napari and vizarr, if the software that is reading the format only supports certain combinations of downsampling factors and offsets, it can just return an error if it receives an unsupported transformation.

Reply:
+1 for requiring a `coordinateTransformation` for ever downsampled image. _Downsampled images are just images,_ so they warrant the same precise metadata as regular images. 

Inferring the coordinate grid of a downsampled image simply from its shape (or the ratio of its dimensions relative to some other image) is very brittle and should be avoided, given that we can just explicitly define the coordinate grid for each downsampled image.

Reply:
OK, so just to clarify. The `coordinateTransformation` `MUST` be on every `datasets` dict but `MAY` also be on the `multiscales` dict?
And if I don't have any pixel-size info or translation, then the first `datasets` will have a no-op transformations list. (e.g. `"coordinateTransformations": []`)? Or can we omit the `coordinateTransformations` for the first `dataset`?

Reply:
Fully agree with @jbms and @d-v-b here; for `napari` / `vizarr` we can for now throw an error if an unavailable transformation is passed. For clarification @will-moore:

> The coordinateTransformation MUST be on every datasets

Yes, I will introduce this constraint later and additionally require that each of these `coordinateTransformations` MUST contain a `scale` transform (and optionally a `translation`). This is the only way to have an unambiguous way to specify the downscaling factors. And for the simplest use case of a single image without resolution information, one can use `scale=[1.0, 1.0 , 1.0]` in the first dataset.

> `MAY` also be on the `multiscales` dict?

This is still up for discussion, and was actually the point I raised initially; I will first change the dataset `coordinateTransform` spec, where we have strong agreement and then open the discussion about this again and also give the pros/cons from my point of view.

> And if I don't have any pixel-size info or translation, then the first `datasets` will have a no-op transformations list. (e.g. `"coordinateTransformations": []`)? Or can we omit the `coordinateTransformations` for the first `dataset`?

In this case, use `scale=[1.0, 1.0, 1.0]`. (More precisely `scale=[1.0] * number_of_axes` (python notation)).

Reply:
Dataset transformations are mandatory now and need to contain scale 0241254. (Please let me know if you have very big concerns about this @joshmoore @sbesson or @will-moore, but from a developer perspective this is a much better choice than leaving this implicit; and of course comment if this still needs to be clarified further in the spec.)

Reply:
> In this case, use scale=[1.0, 1.0, 1.0]. (More precisely scale=[1.0] * number_of_axes (python notation)).

This convention works for me. I expect in that case, the corresponding spatial axes should have no `unit` element, otherwise this should be interpretated as a pixel/voxel size of 1 micron/1 mm/...